addons/product/models/product_template.py

```diff
@@ -873,7 +877,7 @@ def _get_variant_for_combination(self, combination):
         return self.env['product.product'].browse(self._get_variant_id_for_combination(attribute_values))
 
     @api.multi
-    @tools.ormcache('self', 'attribute_values')
+    @tools.ormcache('self.id', 'attribute_values.ids')
```

Comment:
The parameters given to `ormcache` specify how to build a *key* for the cache.
Therefore it must be hashable.
```suggestion
    @tools.ormcache('self.id', 'frozenset(attribute_values.ids)')
```

Reply:
Got it, I made the change for `frozenset`.